src/main/webapp/app/entities/quiz-exercise/quiz-exercise-detail.component.ts

```diff
@@ -176,6 +178,21 @@ export class QuizExerciseDetailComponent implements OnInit, OnChanges, OnDestroy
         return 'isVisibleBeforeStart';
     }
 
+    canDeactivate(): Observable<boolean> | boolean {
+        if (this.pendingChanges()) {
```

Comment:
Can we write it like that?
```suggestion
        return this.pendingChanges();
```

Reply:
Please check again to replace the `canDeactivate` function with
```ts
return !this.pendingChanges();
```

Reply:
Done

Reply:
Without the `if (this.pendingChanges())`!